### ISSUE TITLE
Parts excluded by ifdef should show fewer validation errors

### ISSUE THREAD

**Issue by ahus1** [asciidoctor/asciidoctor-intellij-plugin#1150]
## Why the new feature should be added

In some setups parts are excluded by preprocessor macros, and those parts would never validate when shown in an ide. 

The example below should not show an error for the missing file, and no warning for the undefined attribute. 

```
:anattribute!:  
ifdef::anattribute[]  
xref:nonexistent-no-error.adoc[]  
{nonexistentattribute-no-warning}
endif::[]
xref:nonexistent-show-error.adoc[]  
{nonexistentattribute-show-warning}
```

## How the new feature should work

* If the attribute is not defined as above, and never set to a value anywhere else in the context, it would be ok to assume that that part would never be rendered
* The plugin should then suspend the warnings for those parts in the block. 

A first naive implementation: 

* When validating, search upwards. 
* If first find an endif, assume its outside of a block, and continue validation
* If first find is an ifdef, evaluate it, and if it doesn't match assume it is excluded and not continue with validation

**Comment by ahus1** [comment]
An example project is available here: https://github.com/debezium/debezium/tree/main/documentation

**Comment by ahus1** [comment]
Before with 0.37.45 - `xref:[]`s are marked as broken, attributes are marked as undefined: 

![image](https://user-images.githubusercontent.com/3957921/189485120-83ccf6dc-9b1d-4026-982e-e37ce341e95c.png)

![image](https://user-images.githubusercontent.com/3957921/189485142-d8bbd23e-c4cd-4d20-88e4-885b84e4a9ab.png)

After with 0.37.46 - `xref:[]`s in never rendered blocks are not marked as broken they don't resolve, and not defined attributes are not highlighted as warnings: 

![image](https://user-images.githubusercontent.com/3957921/189485418-a5373493-bcea-4942-9dcf-3ee5c1c2a4e4.png)

![image](https://user-images.githubusercontent.com/3957921/189485439-f8eb58d4-f158-4ef5-9a40-5a0bb3c3981c.png)

**Comment by ahus1** [comment]
Hi @Naros - I noticed that the AsciiDoc plugin for IntelliJ show a lot of warnings and errors when reviewing the Antora content due to the many ifdef's that contain content that is never rendered in Antora. 

Starting with version 0.37.46 (currently available as a pre-release) it hides such errors and warnings as they are not relevant in that context. See the previous comment in this issue for some examples.

The Debezium docs contain a lot of source code snippets are incomplete due to the fact that they are excerpts. Due to that the IntelliJ syntax highlighting shows a lot of errors and warnings. To avoid this, users can disable the error highlighting for source blocks for specific languages, or even for all languages (see the end of this note for a screen shot). This feature has been around for some releases, and 0.37.46 fixes a problem that this stopped working in 2022.2 (#1151).

Please let me know if this is helpful for your work. With fewer false positives, it should be simpler to spot real problems. 

The pre-release of the plugin is available from [GitHub releases](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/installing-a-preview-release.html) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).

// cc: @gunnarmorling

![image](https://user-images.githubusercontent.com/3957921/189486021-b4987d19-d493-44ef-908e-00ac21032f20.png)

**Comment by ahus1** [comment]
This change is now available as a regular release.

**Comment by ahus1** [comment]
Unfortunately this change introduced a performance regression. That regression is now fixed in plugin version 0.37.52, please update.
